DNN Platform/Library/Services/Mail/Mail.cs

```diff
@@ -309,7 +309,31 @@ public static string SendMail(int portalId, int userId, MessageType msgType, Por
                 smtpAuthentication,
                 smtpUsername,
                 smtpPassword,
-                smtpEnableSSL);
+                smtpEnableSSL,
+                Host.SMTPAuthProvider);
+        }
+
+        public static string SendMail(string mailFrom, string mailTo, string cc, string bcc, MailPriority priority, string subject, MailFormat bodyFormat, Encoding bodyEncoding, string body,

```

Comment:
Should we really continue pushing more and more complex public API's for this?  Or should the specific implementation be hidden?

Reply:
this parameter is needed for the "test smtp configuration" function, we may consider to refactor this class later.